mne/morph.py

```diff
@@ -0,0 +1,934 @@
+        Path to source subject's anatomical MRI or Nifti1Image
+    mri_to : string | Nifti1Image
+        Path to destination subject's anatomical MRI or Nifti1Image
+    niter_affine : tuple of int
+        Number of levels (``niter_affine.__len__()``) and number per level of
+        iterations to perform the affine transform. Increasing index values
+        for the tuple mean later levels and each int represents the number
+        of iterations in that level.
+    niter_sdr : tuple of int
+        Number of levels (``niter_sdr.__len__()``) and number per level of
+        iterations to perform the sdr transform. Increasing index values
+        for the tuple mean later levels and each int represents the number
+        of iterations in that level.
+    grid_spacing : tuple
+        Voxel size of volume for each spatial dimension.
+        If grid_spacing is None, MRIs won't be resliced. Note that in this case
+        both volumes must have the same number of slices in every
+        spatial dimension.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see :func:`mne.verbose`
+        and :ref:`Logging documentation <tut_logging>` for more).
+
+    Returns
+    -------
+    morph : list of AffineMap and DiffeomorphicMap
+        Affine and Diffeomorphic registration
+
+    Notes
+    -----
+    This function will be used to morph VolSourceEstimate based on an
+    affine transformation and a nonlinear morph, estimated based on
+    respective transformation from the subject's anatomical T1 (brain) to
+    a destination subject's anatomical T1 (e.g. fsaverage). Afterwards the
+    transformation can be applied. Affine transformations are computed
+    based on the mutual information. This metric relates structural changes
+    in image intensity values. Because different still brains expose high
+    structural similarities this method works quite well to relate
+    corresponding features [1]_.
+    The nonlinear transformations will be performed as Symmetric
+    Diffeomorphic Registration using the cross-correlation metric [2]_.
+
+    References
+    ----------
+    .. [1] Mattes, D., Haynor, D. R., Vesselle, H., Lewellen, T. K., &
+    Eubank, W. (2003). PET-CT image registration in the chest using
+    free-form deformations. IEEE transactions on medical imaging, 22(1),
+    120-128.
+
+    .. [2] Avants, B. B., Epstein, C. L., Grossman, M., & Gee, J. C. (2009).
+    Symmetric Diffeomorphic Image Registration with Cross- Correlation:
+    Evaluating Automated Labeling of Elderly and Neurodegenerative Brain,
+    12(1), 26-41. Asymmetry. Journal of Cognitive Neuroscience 25(9),
+    1477-1492, 2013.
+        """
+
+    try:
+        from dipy.align import imaffine, imwarp, metrics, transforms
+        from dipy.align.reslice import reslice
+        import nibabel as nib
+    except ImportError:
+        raise ImportError(
+            "dipy and nibabel is required compute nonlinear "
```

Comment:
I would throw more explicit error. If dipy is missing, only raise an error for dipy.